### PR TITLE
Add JsonLdContext support

### PR DESCRIPTION
**Pull request by brinxmat**
- Adds JsonLdContext and implementations
- Adds Deserializer for JsonLdContext
- Updates DTO-object mapper

### CODE REVIEW

File: json/src/test/java/no/unit/nva/commons/json/deserialization/Thing.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package no.unit.nva.commons.json.deserialization;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeInfo.Id;
+import no.unit.nva.commons.json.JsonLdContext;
+
+@JsonTypeInfo(use = Id.NAME, property = "type")
+public class Thing {
```
Comment (truhacevkir) [inline json/src/test/java/no/unit/nva/commons/json/deserialization/Thing.java:10]: I guess this one can also be converted to record

**Review by truhacevkir — APPROVED** [review]
Looks like Thing is used for test purposes only, maybe rename to JsonLdTestObject or something that makes more sense :)

File: json/src/main/java/no/unit/nva/commons/json/JsonLdContextUri.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package no.unit.nva.commons.json;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import java.net.URI;
+
+public class JsonLdContextUri implements JsonLdContext {
```
[the remark this answers is not in the document]
Comment (axthosarouris) [reply json/src/main/java/no/unit/nva/commons/json/JsonLdContextUri.java:7]: Pay attention that any Java 11 code will not work. On the other hand, this creates pressure to upgrade to Java 17 everywhere :)

[the remark this answers is not in the document]
Comment (brinxmat) [reply json/src/main/java/no/unit/nva/commons/json/JsonLdContextUri.java]: I think we just do this… :D

File: json/src/test/java/no/unit/nva/commons/json/deserialization/Thing.java (hunk 1 above)
Comment (brinxmat) [reply json/src/test/java/no/unit/nva/commons/json/deserialization/Thing.java]: Apparently not :D

File: json/src/main/java/no/unit/nva/commons/json/ld/JsonLdContextUri.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ * Record to allow representation of a JSON-LD remote context (see example).
+ *
+ * <pre>
+ *     {
+ *         "@context" : "https://example.org/jsonldcontext.jsonld"
+ *     }
+ * </pre>
+ * @param context The URI for the remote JSON-LD context.
+ */
+public record JsonLdContextUri(@JsonValue URI context) implements JsonLdContext {
```
Comment (truhacevkir) [inline json/src/main/java/no/unit/nva/commons/json/ld/JsonLdContextUri.java:16]: Er ikke det vakkert :)

**Review by truhacevkir — APPROVED** [review]

**Review by ketilaa — APPROVED** [review]
Elegant ;)

**Pull request merged**